web/src/main/java/se/inera/intyg/webcert/web/service/facade/impl/certificatefunctions/CertificateSignConfirmationFunctionImpl.java

```diff
@@ -50,13 +53,15 @@ public CertificateSignConfirmationFunctionImpl(UtkastService utkastService,
 
     @Override
     public Optional<ResourceLinkDTO> get(Certificate certificate, WebCertUser webCertUser) {
-        if (!authoritiesHelper.isFeatureActive(AuthoritiesConstants.FEATURE_UNIKT_INTYG, certificate.getMetadata().getType())) {
+        if (isUniqueFeatureActive(certificate)) {
             return Optional.empty();
         }
-        final var personnummer = Personnummer.createPersonnummer(certificate.getMetadata().getPatient().getPersonId().getId()).get();
-        final var existingIntyg = utkastService.checkIfPersonHasExistingIntyg(personnummer, webCertUser, certificate.getMetadata().getId());
-        final var previousIntygMap = existingIntyg.getOrDefault(INTYG_INDICATOR, Collections.emptyMap());
-        if (previousIntygMap.containsKey(certificate.getMetadata().getType())) {
+        final var personnummer = getPersonnummer(certificate);
+        final var existingIntyg = getExistingIntyg(certificate, webCertUser, personnummer);
+        final var previousIntygMap = existingIntygWithStatus(existingIntyg, INTYG_INDICATOR);
+        final var previousUtkastMap = existingIntygWithStatus(existingIntyg, UTKAST_INDICATOR);
+        final var previousReplacedMap = existingIntygWithStatus(existingIntyg, ERSATT_INDICATOR);
+        if (previousIntygExists(certificate, previousIntygMap) && !previousIntygExists(certificate, previousReplacedMap)) {
```

Comment:
Not sure that it makes it clearer this way... Maybe should consider refactoring...

Now you it read like previousIntygexists && !previousIntygExists... and you have to check the arguments to understand. 

Maybe better to keep previousIntygMap.containKey(certificate.getMetadata().getType()). You could maybe put the type in a local variable so you reduce the code.

Reply:
Great suggestion! Updated.